### PR TITLE
fix(ui5-dialog): Texts are no longer blurred in Chromium-based browsers

### PR DESCRIPTION
**Pull request by gmkv**
Fixes: #2270

Bug in Chromium-based browsers (https://bugs.chromium.org/p/chromium/issues/detail?id=783388)
causes elements with transform translate applied to be blurred
due to values ending up resulting in fractional values.

As a workaround, ui5-dialog is now centered using position fixed (with top and left).

### CODE REVIEW

File: packages/main/src/Dialog.js [hunk 1]
```diff
@@ -212,10 +215,43 @@ class Dialog extends Popup {
...
+		delete this._dragMouseUpHandler;
+		delete this._resizeMouseMoveHandler;
+		delete this._resizeMouseUpHandler;
+	}
+
+	show() {
+		ResizeHandler.register(this, this._screenResizeHandler);
+		ResizeHandler.register(document.body, this._screenResizeHandler);
+
+		this.style.visibility = "hidden";
```
Comment (dimovpetar) [inline packages/main/src/Dialog.js:229]: This doesn't seem to do anything. Why is it needed?

File: packages/main/src/Dialog.js [hunk 2]
```diff
@@ -212,10 +215,43 @@ class Dialog extends Popup {
 	}
 
 	onExitDOM() {
```
Comment (dimovpetar) [inline packages/main/src/Dialog.js:217]: is it really needed to delete the handlers? Can't we let them be and initialize them in the constructor()?

**Review by dimovpetar — CHANGES_REQUESTED** [review]
1) When draggable dialog is dragged, then resizing the window shouldn't recenter the dialog
2) When resizable dialog is resized, also resizing the window shouldn't recenter the dialog

This is the behavior of the OpenUI5 Dialog https://openui5.hana.ondemand.com/entity/sap.m.Dialog/sample/sap.m.sample.Dialog

File: packages/main/src/Dialog.js [hunk 3]
```diff
@@ -212,10 +215,43 @@ class Dialog extends Popup {
...
-		this._dragMouseMoveHandler = null;
-		this._dragMouseUpHandler = null;
+		delete this._screenResizeHandler;
+		delete this._dragMouseMoveHandler;
+		delete this._dragMouseUpHandler;
+		delete this._resizeMouseMoveHandler;
+		delete this._resizeMouseUpHandler;
+	}
```
Comment (dimovpetar) [inline packages/main/src/Dialog.js:223]: is it really needed to delete the handlers? Can't we let them be and initialize them in the constructor()?

Comment (gmkv) [reply packages/main/src/Dialog.js:223]: Deleting isn't really needed, you're right. With regards to when should the handlers be initialized, I want to delay this until the web component is going to be rendered on the page. Contructors get called before connectedCallback which is too early. Here, connectedCallback is onEnterDOM.

File: packages/main/src/Dialog.js (hunk 1 above)
Comment (gmkv) [reply packages/main/src/Dialog.js:229]: Removed it

**Comment by gmkv** [comment]
> 1. When draggable dialog is dragged, then resizing the window shouldn't recenter the dialog
> 2. When resizable dialog is resized, also resizing the window shouldn't recenter the dialog
> 
> This is the behavior of the OpenUI5 Dialog https://openui5.hana.ondemand.com/entity/sap.m.Dialog/sample/sap.m.sample.Dialog

Waiting for a response from UX colleagues regarding this.

File: packages/main/src/Dialog.js [hunk 4]
```diff
@@ -390,6 +402,8 @@ class Dialog extends Popup {
 	}
 
 	_attachResizeHandlers() {
+		ResizeHandler.deregister(this, this._screenResizeHandler);
+
```
Comment (TeodorTaushanov) [inline packages/main/src/Dialog.js:406]: Add also 
ResizeHandler.deregister(document.body, this._screenResizeHandler);

File: packages/main/src/Dialog.js [hunk 5]
```diff
@@ -201,21 +214,32 @@ class Dialog extends Popup {
 		this._isRTL = this.effectiveDir === "rtl";
 		this.onPhone = isPhone();
 		this.onDesktop = isDesktop();
+
+		ResizeHandler.deregister(this, this._screenResizeHandler);
```
Comment (TeodorTaushanov) [inline packages/main/src/Dialog.js:218]: Register/deregister these events in onEnterDOM/onExitDOM methods

	onEnterDOM() {
		ResizeHandler.register(this, this._screenResizeHandler);
		ResizeHandler.register(document.body, this._screenResizeHandler);
	}

	onExitDOM() {
		ResizeHandler.deregister(this, this._screenResizeHandler);
		ResizeHandler.deregister(document.body, this._screenResizeHandler);
	}

Comment (gmkv) [reply packages/main/src/Dialog.js]: Done

File: packages/main/src/Dialog.js (hunk 4 above)
Comment (gmkv) [reply packages/main/src/Dialog.js:421]: Done. Also did the same for the Drag events. Now, the webcomponents acts the same as the OpenUI5 Dialog control with regards to the above comments.

**Review by TeodorTaushanov — APPROVED** [review]

**Review by dimovpetar — APPROVED** [review]

File: packages/main/src/Dialog.js [hunk 6]
```diff
@@ -204,16 +222,33 @@ class Dialog extends Popup {
...
-		this._resizeMouseMoveHandler = this._onResizeMouseMove.bind(this);
-		this._resizeMouseUpHandler = this._onResizeMouseUp.bind(this);
+		ResizeHandler.register(this, this._screenResizeHandler);
+		ResizeHandler.register(document.body, this._screenResizeHandler);
 	}
 
 	onExitDOM() {
-		this._dragMouseMoveHandler = null;
-		this._dragMouseUpHandler = null;
+		ResizeHandler.deregister(this, this._screenResizeHandler);
```
Comment (vladitasev) [inline packages/main/src/Dialog.js:230]: This and the next line are repeated several times in the code - not a big deal but maybe move them to a separate function?

**Review by vladitasev — APPROVED** [review]

**Pull request merged**